core/src/main/kotlin/org/evomaster/core/search/gene/ArrayGene.kt

```diff
@@ -65,4 +65,8 @@ class ArrayGene<T>(
     override fun flatView(): List<Gene> {
         return listOf(this).plus(elements.flatMap { g -> g.flatView() })
     }
+
+    override fun flatViewWithTypeFilter(predicate: (Gene) -> Boolean): List<Gene>{
```

Comment:
furthermore, instead of having and maintaining two different functions (ie flatView and flatViewWithTypeFilter), could keep a single one, with a default predicate, eg, something like:
 flatView(predicate: (Gene) -> Boolean = {true})

Reply:
Yes. true. I will modify like this.